### DISCUSSION TITLE
Representing SMIRKS reaction into Rdkit form

### DISCUSSION THREAD

**Discussion by TAEUKHAM** [rdkit/rdkit#5081] [Q&A]
Hi everyone, Ive recently found out the reaction database consisted of SMIRKS. And I'm trying to convert SMIRK reaction notation into Rdkit style(in form which rdkit could understandable) but having difficulties. Would be glad to know if there is a program or rdkit library that makes it possible. Also Im wondering that if this transformation works in a canonical(one-to-one) way. Thanks!

**Comment by greglandrum** [comment]
The RDKit's `ReactionFromSmarts()` can often correctly handle SMIRKS.
If you can share one or two reactions you're having problems with I can try and help

**Reply by TAEUKHAM** [reply]
Thanks for the reply @greglandrum. Currently I'm trying to extract smirk reactions from compound set which is comprised of (substrate , metabolite) pair. Here, substrate is turned into metabolite with certain reaction. Ive found a library called 'rdChiral' which is made above rdkit that extracts reaction from pair of chemical compound but unfortunately in only SMART format. So I was looking for method either makes smart reaction to SMIRK reaction or directly extract smirk reaction from pair of chemical compounds.

For example from this pair 
`
substrate : "InChI=1S/C12H14N2O/c1-15-8-2-3-11-10(6-8)9-4-5-13-7-12(9)14-11/h2-3,6,13-14H,4-5,7H2,1H3"
metabolite : "InChI=1S/C11H12N2O/c14-7-1-2-10-9(5-7)8-3-4-12-6-11(8)13-10/h1-2,5,12-14H,3-4,6H2"

==>(want to get smirk reaction like this) [H:1][*:2][#6:3](=,:[*:4][H:5])-[#8;X2R0:6][C:7]([H:8])([H])[H:9]>>[H:8][#6:7]([H:9])=O.[H:1][*:2][#6:3](=,:[*:4][H:5])-[#8;X2R0:6][H]
`
